lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -0,0 +1,227 @@
+
+    def perform_authentication(self, request):
+        """
+        Ensures that the user is authenticated (e.g. not an AnonymousUser), unless DEBUG mode is enabled.
+        """
+        super(CourseViewMixin, self).perform_authentication(request)
+        if request.user.is_anonymous() and not settings.DEBUG:
+            raise AuthenticationFailed
+
+
+class CourseList(CourseViewMixin, ListAPIView):
+    """
+    **Use Case**
+
+        CourseList returns paginated list of courses in the edX Platform. The list can be
+        filtered by course_id
+
+    **Example Request**
+
+          GET /
+          GET /?course_id={course_id1},{course_id2}
+
+    **Response Values**
+
+        * category: The type of content. In this case, the value is always "course".
+
+        * name: The name of the course.
+
+        * uri: The URI to use to get details of the course.
+
+        * course: The course number.
+
+        * due:  The due date. For courses, the value is always null.
+
+        * org: The organization specified for the course.
+
+        * id: The unique identifier for the course.
+    """
+    paginate_by = 10
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = serializers.CourseSerializer
+
+    def get_queryset(self):
+        course_ids = self.request.QUERY_PARAMS.get('course_id', None)
+
+        course_descriptors = []
+        if course_ids:
+            course_ids = course_ids.split(',')
+            for course_id in course_ids:
+                course_key = CourseKey.from_string(course_id)
+                course_descriptor = courses.get_course(course_key)
+                course_descriptors.append(course_descriptor)
+        else:
+            course_descriptors = modulestore().get_courses()
+
+        results = [course for course in course_descriptors if self.user_can_access_course(self.request.user, course)]
+
+        # Sort the results in a predictable manner.
+        results.sort(key=lambda x: x.id)
+
+        return results
+
+
+class CourseDetail(CourseViewMixin, RetrieveAPIView):
+    """
+    **Use Case**
+
+        CourseDetail returns details for a course.
+
+    **Example requests**:
+
+        GET /{course_id}/
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * name: The name of the course.
+
+        * uri: The URI to use to get details of the course.
+
+        * course: The course number.
+
+        * due:  The due date. For courses, the value is always null.
+
+        * org: The organization specified for the course.
+
+        * id: The unique identifier for the course.
+    """
+
+    serializer_class = serializers.CourseSerializer
+
+    def get_object(self, queryset=None):
+        return self.get_course_or_404()
+
+
+class CourseStructure(CourseViewMixin, RetrieveAPIView):
+    """
+    **Use Case**
+
+        Retrieves course structure.
+
+    **Example requests**:
+
+        GET /{course_id}/structure/
```

Comment:
Oh, I misread the urls.py file. Our API design guidelines suggest that we stick with URLS of the form `api/<api_name>/<version>/<resource>/<id>`, which this doesn't match.